### PR TITLE
OWLS-79994: When scaling down the cluster with WLDF the number of replicas will go under the MinNumberOfServer

### PR DESCRIPTION
**Pull request by lennyphan**
When a dynamic cluster is configured, prevent scale down below the minimum dynamic cluster size.  When a replica request below the configured minimum dynamic cluster size, we log the following warning (analogous to the logging done for a request above the configured maxmimum cluster size):

{"timestamp":"03-24-2020T20:24:08.497+0000","thread":12,"fiber":"engine-operator-thread-2-fiber-922","domainUID":"sample-domain1","level":"WARNING","class":"oracle.kubernetes.operator.steps.ManagedServersUpStep$ServersUpStepFactory","method":"logIfReplicasLessThanClusterServersMin","timeInMillis":1585081448497,"message":"Replica request of 0 less than the minimum dynamic server count of 1 configured for cluster cluster-1","exception":"","code":"","headers":{},"body":""}

This commit also updates the 'Cluster Status' (in the domain description) with the configured minimum 'replicas' value:

e.g.

Status:
  Clusters:
    Cluster Name:      cluster-1
    Maximum Replicas:  5
    Minimum Replicas:  1
    Ready Replicas:    2

### CODE REVIEW

File: operator/src/test/java/oracle/kubernetes/operator/wlsconfig/WlsClusterConfigTest.java [hunk 1]
```diff
@@ -241,17 +248,16 @@ public void verifyMaxClusterSizeForDynamicCluster() {
...
     wlsClusterConfig.addServerConfig(createWlsServerConfig("ms-0", 8011, null));
 
     assertThat(wlsClusterConfig.getMaxClusterSize(), equalTo(2));
   }
 
-  @Test
   public void verifyGetUpdateDynamicClusterSizeUrlIncludesClusterName() {
     WlsDynamicServersConfig wlsDynamicServersConfig =
-        createDynamicServersConfig(1, 1, "ms-", "cluster1");
+        createDynamicServersConfig(1, 1, 1,"ms-", "cluster1");
```
Comment (markxnelson) [inline operator/src/test/java/oracle/kubernetes/operator/wlsconfig/WlsClusterConfigTest.java]: space after comma - weird that checkstyle did not get that

File: operator/src/main/java/oracle/kubernetes/operator/wlsconfig/WlsClusterConfig.java [hunk 2]
```diff
@@ -163,6 +163,14 @@ public synchronized int getMaxClusterSize() {
     return hasDynamicServers() ? getClusterSize() + getMaxDynamicClusterSize() : getClusterSize();
   }
 
+  /**
+   * Returns the minimum size of the cluster.
+   * @return  For static clusters, the minimum size can be 0.  Dynamic servers will return the configured value.
+   */
+  public synchronized int getMinClusterSize() {
```
[the remark this answers is not in the document]
Comment (rjeberhard) [reply operator/src/main/java/oracle/kubernetes/operator/wlsconfig/WlsClusterConfig.java]: Agreed... Looks unnecessary.  What are we missing, @lennyphan ?

File: operator/src/test/java/oracle/kubernetes/operator/wlsconfig/WlsClusterConfigTest.java (hunk 1 above)
Comment (rjeberhard) [reply operator/src/test/java/oracle/kubernetes/operator/wlsconfig/WlsClusterConfigTest.java]: I'll check... I wonder if Checkstyle is somehow not running again for test code.

Comment (lennyphan) [reply operator/src/test/java/oracle/kubernetes/operator/wlsconfig/WlsClusterConfigTest.java]: added space after comma after minClusterSize parameter for all createDynamicServersConfig() methods.

File: operator/src/main/java/oracle/kubernetes/operator/wlsconfig/WlsClusterConfig.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (lennyphan) [reply operator/src/main/java/oracle/kubernetes/operator/wlsconfig/WlsClusterConfig.java]: 'synchronized' unnecessary.  My mistake from cut-paste and just overlooked it by accident.  
Thanks.

**Pull request merged**